rotkehlchen/chain/evm/decoding/interfaces.py

```diff
@@ -169,16 +174,34 @@ def __init__(  # pylint: disable=super-init-not-called
 
     def _decode_vote_cast(self, context: DecoderContext) -> DecodingOutput:
         """Decodes a vote cast event"""
-        if context.tx_log.topics[0] not in (VOTE_CAST, VOTE_CAST_WITH_PARAMS):
+        if context.tx_log.topics[0] == VOTE_CAST:
+            event_abi = VOTE_CAST_ABI
+        elif context.tx_log.topics[0] == VOTE_CAST_WITH_PARAMS:
+            event_abi = VOTE_CAST_WITH_PARAMS_ABI
+        else:
             return DEFAULT_DECODING_OUTPUT  # for params event is same + params argument. Ignore it
 
         voter_address = hex_or_bytes_to_address(context.tx_log.topics[1])
         if not self.base.is_tracked(voter_address):
             return DEFAULT_DECODING_OUTPUT
 
-        proposal_id = str(hex_or_bytes_to_int(context.tx_log.data[:32]))
-        supports = bool(hex_or_bytes_to_int(context.tx_log.data[32:64]))
-        notes = f'Voted {"FOR" if supports else "AGAINST"} {self.protocol} governance proposal {self.proposals_url}/{proposal_id}'  # noqa: E501
+        try:
+            _, decoded_data = decode_event_data_abi_str(context.tx_log, event_abi)
+        except DeserializationError as e:
+            log.error(
+                f'Failed to deserialize Juicebox event at '
+                f'{context.transaction.tx_hash.hex()} due to {e}',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        proposal_id = decoded_data[0]
+        supports = decoded_data[1]
+
+        notes_reason = ''
+        if len(decoded_data[3]) != 0:
```

Comment:
had to use the ABI cause this is variable length and hard to do manually?

Reply:
yes, couldn't get it working manually